libmscore/page.cpp

```diff
@@ -318,9 +318,15 @@ QString Page::replaceTextMacros(const QString& s) const
             QChar nc = s[i+1];
             switch(nc.toLatin1()) {
             case 'p': // not on first page 1
-                if (_no) // FALLTHROUGH
+                if (!_no) {
+                    break;
+                }
+                // FALLTHROUGH
```

Comment:
To be honest I don't see how and where that syntax would be any clearer?
It might work better with uncrustify, but it certainly is not clear at all

Reply:
The old syntax relies on very obscure compiler behaviour. At best, I will agree that the old syntax made it clear what was *supposed* to happen, but it was not at all clear that the compiler would actually do it that way.

```c++
case 'p':
    if (_no) // FALLTHROUGH
case 'N':
    doSomething();
```

If `_no` evaluates to `true` then the comment makes the *intended* behaviour clear enough, but it is not clear that the compiler will actually follow that instruction (it is a comment, after all).

If `_no` is `false` then it is completely unclear what will happen. Does it fall through anyway? If not, then we must assume that `case 'N'` somehow belongs to the previous `if` statement. But then what happens if `case 'N'` is encountered on its own?

Uncrustify interpreted the snippet as follows:

```c++
case 'p':
    if (_no) {
        // FALLTHROUGH
        case 'N':
            doSomething();
    }
```

But now `case 'N'` can never be reached except via `case 'p'`.

My new syntax is as follows:

```c++
case 'p':
    if (!_no) {
        break;
    }
    // FALLTHROUGH
case 'N':
    doSomething();
```

This makes it clear that `case 'N'` and `case 'p'` are two separate cases. It also makes it clear what will happen for `case 'p'` regardless of whether `_no` is `true`.

Reply:
The fallthrough comment is just a comment, but also prevent compiler warning.

Reply:
Of course, but that doesn't change the fact that the old syntax is basically a neat trick that exploits a strange quirk of compiler logic. It works, but it doesn't look right. Anyway, we need to change it if we want it to work with Uncrustify. This was the only thing preventing us from testing for code style compliance on Travis CI.

Reply:
Yes, if uncrustify needs that, we should (and need to) do it. But the code is not clearer...